javatests/arcs/sdk/HandleHolderBaseTest.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.sdk
+
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class HandleHolderBaseTest {
+
+  @Test
+  fun handles_accessMissingHandle_throwsNoElemException() {
+    val handleHolder = DummyHandleHolder()
+
+    assertFailsWith<NoSuchElementException> {
+      handleHolder.readHandle
+    }.also {
+      assertThat(it.message).isEqualTo(
+        "Handle readHandle has not been initialized in TestParticle yet."
+      )
+    }
+  }
+
+  @Test
+  fun handles_nameNotInEntitySpec_throwsNoElemException() {
```

Comment:
Similar for this test: it's only testing the structure that was created in this test.